### PR TITLE
[express-unless] Loosen too restrictive typings for path property, add missing ext property

### PR DESCRIPTION
**Pull request by michal-b-kaminski**
Previous pull request (#32549) restricted types for `Options.path` property too much. This PR adds other possible types. 
Additionaly, adds missing `Options.useOriginalUrl` property.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/jfromaniello/express-unless#current-options
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Comment by michal-b-kaminski** [comment]
@joaovieira Could you please review #32840? Original owner doesn't appear to be active.

Your last change restricted allowed types for `Options.path` property too much. This PR adds remaining possible types.

File: types/express-unless/index.d.ts [hunk 1]
```diff
@@ -11,11 +12,14 @@ declare function unless(options: unless.Options): express.RequestHandler;
 declare function unless(options: unless.Options["custom"]): express.RequestHandler;
 
 declare namespace unless {
+    type pathFilter = string | RegExp | { url: string | RegExp, methods?: string[] };
```
Comment (joaovieira) [inline types/express-unless/index.d.ts]: the object form can also have `method?: string | string[]` like the top-level option
https://github.com/jfromaniello/express-unless/blob/master/index.js#L25

Comment (joaovieira) [reply types/express-unless/index.d.ts]: `url` can also be a `RegExp` https://github.com/jfromaniello/express-unless/blob/master/index.js#L69

**Review by joaovieira — CHANGES_REQUESTED** [review]
@michal-b-kaminski Sorry and thanks for fixing it and adding tests. Looking at the source code there's another edge case.

File: types/express-unless/index.d.ts (hunk 1 above)
Comment (michal-b-kaminski) [reply types/express-unless/index.d.ts]: You are right about `method` optional property in object form, I will fix that. But what do you mean in your second comment? In my declarations there is:
`type pathFilter = string | RegExp | { url: string | RegExp, methods?: string[] };`
So in object form `url` can either be a `string` or a `RegExp`.

**Review by joaovieira — APPROVED** [review]

**Review by joaovieira — APPROVED** [review]

**Comment by joaovieira** [comment]
@typescript-bot @armanio123 this PR has been approved 7 days ago and inactive for 4. Anyone willing to merge this?

**Comment by armanio123** [comment]
@joaovieira For some reason I don't know yet, the typescript-bot didn't recognize you as a type owner. Merging it right now.
